pkg/util/test/util.go

```diff
@@ -505,6 +545,11 @@ func expectedPods(cluster *corev1.StorageCluster) (int, error) {
 		dummyPod.Spec.Affinity = &v1.Affinity{
 			NodeAffinity: cluster.Spec.Placement.NodeAffinity.DeepCopy(),
 		}
+	} else {
+		newCluster := addDefaultNodeAffinityRules(cluster)
```

Comment:
You are setting the node affinity on the passed cluster object as well as returning the same cluster object.
Do you want to change the passed cluster object or just return a new one with affinity rules? 
If you don't want to change the passed cluster object, why not just return the nodeaffinity object here and use that?

Reply:
Done. I'm now using that func to create a Node Affinity object, without passing cluster to the func or passing the object back to the cluster.